.gitlab-ci.yml

```diff
@@ -0,0 +1,141 @@
+    - 'declare BUNDLER_INSTALL_CMD=(bundle install --no-binstubs --jobs $(nproc) "${FLAGS[@]}")'
+    - 'mkdir -p ${GEM_HOME} ${BUNDLER_BIN}'
+    - 'gem list -ie "${GEM_BUNDLER_VER[@]}" --silent bundler || "${GEM_INSTALL_CMD[@]}" --local "${GEM_BUNDLER_VER[@]}" bundler || "${GEM_INSTALL_CMD[@]}" "${GEM_BUNDLER_VER[@]}" bundler'
+    - 'rm -rf pkg/ || :'
+    - 'bundle check || rm -f Gemfile.lock && ("${BUNDLER_INSTALL_CMD[@]}" --local || "${BUNDLER_INSTALL_CMD[@]}" || bundle pristine ||  "${BUNDLER_INSTALL_CMD[@]}") || { echo "PIPELINE: Bundler could not install everything (see log output above)" && exit 99 ; }'
+
+
+    # Diagnostic bundler, ruby, and gem checks:
+    - 'bundle exec rvm ls || :'
+    - 'bundle exec which ruby || :'
+    - 'bundle show sync || :'
+    - 'bundle exec gem list sync || :'
+
+    # install yaml-lint for the linting phase:
+    - 'gem install --conservative --minimal-deps yaml-lint'
+
+.rubocop_project_dir: &rubocop_project_dir
+  cache:
+    untracked: true
+  before_script:
+    # install rubocop:
+    - 'gem install --conservative --minimal-deps rubocop'
+
+.setup_bolt_env: &setup_bolt_env
+  cache:
+    untracked: true
+  before_script:
+    # install bolt:
+    - 'wget https://apt.puppet.com/puppet-tools-release-xenial.deb'
+    - 'sudo dpkg -i puppet-tools-release-xenial.deb'
+    - 'sudo apt-get update'
+    - 'sudo apt-get install puppet-bolt'
+    - 'command -v bolt || if command -v apt-get; then apt-get update; apt-get install -y bolt; fi ||:'
+
+# Testing Environments
+#-----------------------------------------------------------------------
+
+.lint_tests: &lint_tests
+  stage: 'validation'
+  tags: ['docker']
+  <<: *setup_bundler_env
+  script:
+    - bash -c 'for json in $(find . -type f -name "*.json"); do echo "== $json" && jq . "$json"; done'
+.lint_tests: &lint_tests
+  stage: 'validation'
+  tags: ['docker']
+  <<: *setup_bundler_env
+  script:
+    - bash -c 'find . -type f -name "*.sh"  | xargs shellcheck'
+
+.lint_tests: &lint_tests
+  stage: 'validation'
+  tags: ['docker']
+  <<: *setup_bundler_env
+  script:
+    - bash -c 'find . -type f -name "*.yaml" -o -name "*.yml" | xargs yaml-lint'
+
+.lint_tests: &lint_tests
+  stage: 'validation'
+  tags: ['docker']
+  <<: *setup_bolt_env
+  script:
+    - 'BOLT_DISABLE_ANALYTICS=yes bolt task show | grep releng::'
+
+# Travis CI testing was disabled.
+#-----------------------------------------------------------------------
+#.lint_tests: &lint_tests
+#  stage: 'validation'
+#  tags: ['docker']
+#  <<: *rubocop_project_dir
+#  script:
+#    - 'export PROJECT_DIR="set_travis_env_vars"'
+#    - 'rubocop --fail-level warning --display-only-fail-level-offenses $PROJECT_DIR'
+
+.lint_tests: &lint_tests
+  stage: 'validation'
+  tags: ['docker']
+  <<: *rubocop_project_dir
+  script:
+    - 'export PROJECT_DIR="octokit_repo_puncher"'
+    - 'rubocop --fail-level warning --display-only-fail-level-offenses $PROJECT_DIR'
```

Comment:
This step won't install the project's `Gemfile` dependencies if it uses the keys defined in `&rubocop_project_dir` instead of `&setup_bundler_env`.

This probably wasn't clear from reading the original Travis pipeline—in Travis, bundle executes automatically before running `ruby` builds.